File: algorithms/lua/sieve.lua
local function sieve(n)
    local is_composite = {}
    -- فقط تا 10M برای Lua — بیشتر timeout می‌شه
    for i = 0, n do is_composite[i] = false end
    is_composite[0] = true
    is_composite[1] = true

    local i = 2
    while i * i <= n do
        if not is_composite[i] then
            local j = i * i
            while j <= n do
                is_composite[j] = true
                j = j + i
            end
        end
        i = i + 1
    end

    local count, last = 0, -1
    for k = 2, n do
        if not is_composite[k] then
            count = count + 1
            last  = k
        end
    end
    return count, last
end

local function main()
    local n   = tonumber(io.read("*l"))

    local start         = os.clock()
    local count, last   = sieve(n)
    local ms            = (os.clock() - start) * 1000

    print(count)
    print(last)
    io.stderr:write(string.format("%.3f\n", ms))
end

main()

File: algorithms/rust/sieve.rs
use std::io::{self, Read, Write, BufWriter};
use std::time::Instant;

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    let start = Instant::now();

    let mut is_composite = vec![false; n + 1];
    is_composite[0] = true;
    if n > 0 { is_composite[1] = true; }

    let mut i = 2;
    while i * i <= n {
        if !is_composite[i] {
            let mut j = i * i;
            while j <= n {
                is_composite[j] = true;
                j += i;
            }
        }
        i += 1;
    }

    let mut count = 0u64;
    let mut last  = 0i64;
    for k in 2..=n {
        if !is_composite[k] {
            count += 1;
            last   = k as i64;
        }
    }

    let ms = start.elapsed().as_secs_f64() * 1000.0;

    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    writeln!(out, "{}", count).unwrap();
    writeln!(out, "{}", last).unwrap();
    eprintln!("{:.3}", ms);
}